compute_endpoint/globus_compute_endpoint/engines/helper.py

```diff
@@ -56,19 +61,25 @@ def execute_task(
         uuid.UUID | str | tuple[str, str] | list[TaskTransition] | dict[str, str],
     ]
 
+    task_id_str = str(task_id)
     os.environ.pop("GC_TASK_SANDBOX_DIR", None)
-    os.environ["GC_TASK_UUID"] = str(task_id)
+    os.environ["GC_TASK_UUID"] = task_id_str
 
-    if not run_dir or not os.path.isabs(run_dir):
-        raise RuntimeError(
-            f"execute_task requires an absolute path for run_dir, got {run_dir=}"
+    if result_size_limit < 128:
+        raise ValueError(
+            f"Invalid result limit; must be at least 128 bytes ({result_size_limit=})"
         )
 
-    os.makedirs(run_dir, exist_ok=True)
+    run_dir = pathlib.Path(run_dir)
+    if not run_dir.is_absolute():
+        raise ValueError(f"Absolute path required, not relative: {str(run_dir)}")
+
+    run_dir.mkdir(parents=True, exist_ok=True)
     os.chdir(run_dir)
     if run_in_sandbox:
-        os.makedirs(str(task_id))  # task_id is expected to be unique
-        os.chdir(str(task_id))
+        sb_dir = run_dir / task_id_str
```

Comment:
I'm not sure what the new logic here does. At this point we are in the `run_dir`. Can you clarify what we gain by calling mkdir on the absolute path.

Reply:
There's no value-add per se.  These are all functionally equivalent:

```python
os.chdir("/tmp/")  # with the side-effect of changing the working dir
os.makedirs("a/b/c/")

os.makedirs("/tmp/a/b/c")

pathlib.Path("/tmp/a/b/c").mkdir(parents=False)
```

I was taking the opportunity to migrate to `pathlib.Path` (given `os.PathLike` in the signature) and build upon the now-proven-valid `run_dir`.  If that's too much for this PR, I can pull this out.

Reply:
What do you think about moving the dir creation and switch after the task directory is finalized? Something like this:
``` python
task_dir = run_dir
if run_in_sandbox:
    task_dir = run_dir / task_id_str
task_dir.mkdir(parents=True, exist_ok=True)
os.chdir(task_dir)
```

Reply:
Ooh, that's much nicer.  Thanks for the suggestion!